tests/snmp/memory.py

```diff
@@ -11,28 +11,35 @@ def get_free_memory():
     return free_memory
 
 
-free_memory = get_free_memory()
+reserve_free_memory = 256 * 1024 * 1024    # reserve 256M
+free_memory = get_free_memory()            # KB
+chunk_size = 32 * 1024 * 1024              # chunk size 32M
 total_chars = 512000000
 
-if (free_memory * 1024) > int(total_chars * 130 / 100):
+# reserve 256M for system run
+if (free_memory * 1024 - reserve_free_memory) > total_chars:
     load = []
     count = 1000
     for i in range(0, count):
         load.append([' ' * int(total_chars / count)])
         time.sleep(0.01)
 else:
-    chunk_size = int((free_memory * 1024 * 100 / 130) / 2)
-    large_string = ""
-    remaining_chars = total_chars
-    # print("Free Memory: {} total_chars {} chunk_size {}".format(free_memory, total_chars, chunk_size))
-
-    try:
-        while remaining_chars > 0:
-            chunk = ' ' * min(chunk_size, remaining_chars)
-            large_string += chunk
-            remaining_chars -= chunk_size
-
-        for i in range(0, len(large_string), chunk_size):
-            print(large_string[i:i+chunk_size])
-    except MemoryError:
-        print("Not enough memory to generate and print the large string.")
+    # for small memory device, use chunk size instead of total_chars
+    command = "sudo sh -c 'echo 3 > /proc/sys/vm/drop_caches'"
```

Comment:
if we are reserving the required memory, do we still need to drop_caches?

Reply:
Hi, @SuvarnaMeenakshi , thanks for reviewing the PR.
Based on the test result, on 7050qx, normally, the available memory is about 600M, but the free memory is less than 200M which is less than the threshold of the reserve free memory. 
So I added this CMD to drop the caches, thanks.